snapcraft/plugins/python.py

```diff
@@ -443,9 +443,14 @@ def list(self, exec_func=None):
         output = exec_func(cmd, env=self._env)
         package_listing = {}
         version_regex = re.compile('\((.+)\)')
+        version_regex_alt = re.compile('((\d+)\.)?((\d+)\.)?(\*|\d+)')
         for line in output.splitlines():
             line = line.split()
             m = version_regex.search(line[1])
+            if not m:
```

Comment:
version_regex_alt is only used inside this if block, so it would be better if you move the variable assignment here.